backend/src/services/wallet_service.py

```
@@ -50,6 +51,27 @@ def _initialize_wallet(self):
+        try:
+            mint_pubkey = Pubkey.from_string(mint_address)
+            account_info = self.solana_client.get_account_info(mint_pubkey)
+            if account_info.value:
+                # Basic decoding of mint account to get decimals
+                # The decimals are at offset 44 in the mint account data
+                data = account_info.value.data
+                if len(data) >= 45:
+                    decimals = data[44]
+                    self.token_decimals_cache[mint_address] = decimals
+                    return decimals
+        except Exception as e:
```

Ruff (0.15.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)